src/mongo/db/storage/wiredtiger/encryption_keydb.cpp

```diff
@@ -413,7 +420,16 @@ void EncryptionKeyDB::clone(EncryptionKeyDB *old) {
 }
 
 void EncryptionKeyDB::store_masterkey() {
-    vaultWriteKey(base64::encode(StringData{(const char*)_masterkey, _key_len}));
+    auto encodedKey = base64::encode(StringData{(const char*)_masterkey, _key_len});
+    if (!encryptionGlobalParams.kmipServerName.empty()) {
+        kmipWriteKey(encodedKey);
+    } else if (!encryptionGlobalParams.vaultServerName.empty()) {
+        vaultWriteKey(encodedKey);
+    } else {
+        std::logic_error(
```

Comment:
Sorry for intrusion, just caught my eye
```suggestion
        throw std::logic_error(
```

Reply:
Fixed in https://github.com/percona/percona-server-mongodb/pull/844